### PR TITLE
Test updates for assessment resource links

### PR DESCRIPTION
**Pull request by alxdarksage**

### CODE REVIEW

File: src/test/java/org/sagebionetworks/bridge/sdk/integration/AssessmentResourceTest.java [hunk 1]
```diff
@@ -0,0 +1,226 @@
...
+        }
+        
+        revisions = sharedApi.getSharedAssessmentRevisionsById(id, null, null, true).execute().body();
+        for (Assessment oneRevision : revisions.getItems()) {
+            sharedApi.deleteSharedAssessment(oneRevision.getGuid(), true).execute();
+        }
+    }
+
+    @Test
+    public void test() throws Exception {
```
Comment (DwayneJengSage) [inline src/test/java/org/sagebionetworks/bridge/sdk/integration/AssessmentResourceTest.java]: Should we also have a test for deleteAssessmentResource and deleteSharedAssessmentResource for logical deletes?

File: src/test/java/org/sagebionetworks/bridge/sdk/integration/AssessmentResourceTest.java [hunk 2]
```diff
@@ -0,0 +1,226 @@
...
+        } catch(UnauthorizedException e) {
+            // expected
+        }
+
+        // publish the assessment and resource
+        assessmentApi.publishAssessment(assessment.getGuid()).execute().body();
+        assessmentApi.publishAssessmentResource(id, ImmutableList.of(resourceGuid)).execute().body();
+
+        // Resource should be published along with the assessment.
+        PagedExternalResourceList sharedResourcesPage = sharedAssessmentsApi.getSharedAssessmentResources(
```
Comment (DwayneJengSage) [inline src/test/java/org/sagebionetworks/bridge/sdk/integration/AssessmentResourceTest.java]: What about a test for getSharedAssessmentResource() (singlular)?

File: src/test/java/org/sagebionetworks/bridge/sdk/integration/AssessmentResourceTest.java [hunk 3]
```diff
@@ -0,0 +1,226 @@
...
+        Substudy org = subApi.getSubstudy(SUBSTUDY_ID_1).execute().body();
+        if (org == null) {
+            Substudy substudy = new Substudy().id(SUBSTUDY_ID_1).name(SUBSTUDY_ID_1);
+            subApi.createSubstudy(substudy).execute();
+        }
+        org = subApi.getSubstudy(SUBSTUDY_ID_2).execute().body();
+        if (org == null) {
+            Substudy substudy = new Substudy().id(SUBSTUDY_ID_2).name(SUBSTUDY_ID_2);
+            subApi.createSubstudy(substudy).execute();
+        }
```
Comment (DwayneJengSage) [inline src/test/java/org/sagebionetworks/bridge/sdk/integration/AssessmentResourceTest.java]: getSubstudy() throws if the substudy doesn't exist.

File: src/test/java/org/sagebionetworks/bridge/sdk/integration/AssessmentResourceTest.java [hunk 4]
```diff
@@ -0,0 +1,226 @@
...
+
+        developer = new TestUserHelper.Builder(AssessmentTest.class).withRoles(DEVELOPER)
+                .withSubstudyIds(ImmutableSet.of(SUBSTUDY_ID_1)).createAndSignInUser();
+        otherDeveloper = new TestUserHelper.Builder(AssessmentTest.class).withRoles(DEVELOPER)
+                .withSubstudyIds(ImmutableSet.of(SUBSTUDY_ID_2)).createAndSignInUser();
+        assessmentApi = developer.getClient(AssessmentsApi.class);
+        sharedAssessmentsApi = developer.getClient(SharedAssessmentsApi.class);
+        badDevApi = otherDeveloper.getClient(AssessmentsApi.class);
+        id = randomIdentifier(AssessmentTest.class);
```
Comment (DwayneJengSage) [inline src/test/java/org/sagebionetworks/bridge/sdk/integration/AssessmentResourceTest.java]: Should move id to the top of before(). That way, if the test fails in before(), you won't get a weird NullPointerException in after().

File: src/test/java/org/sagebionetworks/bridge/sdk/integration/AssessmentResourceTest.java [hunk 5]
```diff
@@ -0,0 +1,226 @@
...
+        assertFalse(resource.isUpToDate());
+        
+        // update the resource. this should increment the version.
+        resource.setUrl(URL + "2");
+        resource = assessmentApi.updateAssessmentResource(id, resourceGuid, resource).execute().body();
+        assertEquals(URL + "2", resource.getUrl());
+        assertNotEquals(version, resource.getVersion());
+        assertTrue(resource.isUpToDate());
+
+        // test paging values:
```
Comment (DwayneJengSage) [inline src/test/java/org/sagebionetworks/bridge/sdk/integration/AssessmentResourceTest.java]: There's a comment here saying to test paging values, but no actual paging tests.

**Review by DwayneJengSage — CHANGES_REQUESTED** [review]

File: src/test/java/org/sagebionetworks/bridge/sdk/integration/AssessmentResourceTest.java (hunk 1 above)
Comment (alxdarksage) [reply src/test/java/org/sagebionetworks/bridge/sdk/integration/AssessmentResourceTest.java]: Added both of these at the end of the test.

File: src/test/java/org/sagebionetworks/bridge/sdk/integration/AssessmentResourceTest.java (hunk 2 above)
Comment (alxdarksage) [reply src/test/java/org/sagebionetworks/bridge/sdk/integration/AssessmentResourceTest.java]: Added this (it's also exercised as part of verifying that logically deleted items are still accessible if you have a direct reference to them).

File: src/test/java/org/sagebionetworks/bridge/sdk/integration/AssessmentResourceTest.java (hunk 3 above)
Comment (alxdarksage) [reply src/test/java/org/sagebionetworks/bridge/sdk/integration/AssessmentResourceTest.java]: Refactored this to create if the call throws EntityNotFoundException. Tested this by deleting them and running the test. Also noted that otherDeveloper wasn't being cleaned up and not using the right class when creating the test email accounts for the developers, fixed all that.

File: src/test/java/org/sagebionetworks/bridge/sdk/integration/AssessmentResourceTest.java (hunk 4 above)
Comment (alxdarksage) [reply src/test/java/org/sagebionetworks/bridge/sdk/integration/AssessmentResourceTest.java]: moved

File: src/test/java/org/sagebionetworks/bridge/sdk/integration/AssessmentResourceTest.java (hunk 5 above)
Comment (alxdarksage) [reply src/test/java/org/sagebionetworks/bridge/sdk/integration/AssessmentResourceTest.java]: Added tests of the actual paging, not just the filtering arguments (at end of test).

**Review by DwayneJengSage — APPROVED** [review]

**Pull request merged**